robot-core/src/main/java/org/obolibrary/robot/ReasonOperation.java

```diff
@@ -424,6 +440,43 @@ private static void addInferredAxioms(
         }
       }
 
+      if (tautologiesOption.equalsIgnoreCase("structural")) {
+        if (a instanceof OWLSubClassOfAxiom) {
+          OWLSubClassOfAxiom subClassOfAxiom = (OWLSubClassOfAxiom) a;
+          if (subClassOfAxiom.getSuperClass().isOWLThing()) {
+            continue;
+          } else if (subClassOfAxiom.getSubClass().isOWLNothing()) {
+            continue;
+          } else if (subClassOfAxiom.getSubClass().equals(subClassOfAxiom.getSuperClass())) {
+            continue;
+          }
+        } else if (a instanceof OWLEquivalentClassesAxiom) {
+          OWLEquivalentClassesAxiom equivAxiom = (OWLEquivalentClassesAxiom) a;
+          if (equivAxiom.getClassExpressions().size() < 2) {
+            continue;
+          }
+        } else if (a instanceof OWLClassAssertionAxiom) {
+          OWLClassAssertionAxiom classAssertion = (OWLClassAssertionAxiom) a;
+          if (classAssertion.getClassExpression().isOWLThing()) {
+            continue;
+          }
+        } else if (a instanceof OWLObjectPropertyAssertionAxiom) {
+          OWLObjectPropertyAssertionAxiom assertion = (OWLObjectPropertyAssertionAxiom) a;
+          if (assertion.getProperty().isOWLTopObjectProperty()) {
+            continue;
+          }
+        } else if (a instanceof OWLDataPropertyAssertionAxiom) {
+          OWLDataPropertyAssertionAxiom assertion = (OWLDataPropertyAssertionAxiom) a;
+          if (assertion.getProperty().isOWLTopDataProperty()) {
+            continue;
+          }
+        }
+      } else if (tautologiesOption.equalsIgnoreCase("all")) {
+        if (tautologyChecker.isEntailed(a)) {
```

Comment:
If for some reason `OWLOntology empty = OWLManager.createOWLOntologyManager().createOntology();` throws an exception, the `tautologyChecker` will be `null` and this will throw a `NullPointerException`. I'm not sure why that would happen, but you have a catch for it above.

Reply:
@beckyjackson I updated two ways:
- Here I check for null before using.
- Above, I don't catch the `OWLOntologyCreationException` anymore, since this would be a strange error and we shouldn't continue. However it changes the signature of this method; but, the method is *private*. Does that make it okay for binary compatibility?

Reply:
I convinced myself that the backwards compatibility checks are satisfied because the method is private.